pkg/assembler/backends/keyvalue/isDependency.go

```diff
@@ -214,10 +362,13 @@ func (c *demoClient) IsDependency(ctx context.Context, filter *model.IsDependenc
 				if err != nil {
 					return nil, err
 				}
-				out, err = c.addDepIfMatch(ctx, out, filter, link)
+				dep, err := c.depIfMatch(ctx, filter, link)
 				if err != nil {
 					return nil, gqlerror.Errorf("%v :: %v", funcName, err)
 				}
+				if dep != nil {
```

Comment:
without this the e2e failed.

Reply:
Thank you, I had no idea what was going wrong!